src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "DDEV Name Resolution, Wildcards, Working Offline"
+pubDate: 2024-03-27
+#modifiedDate: 2024-03-03
+summary: How hostname name resolution works in a DDEV environment, including how wildcards work
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/03/2024-ddev.png
+  alt: Hostname resolution with DDEV
+  credit: 'Ideogram.ai: the words "2024" and "DDEV" next to each other'
+categories:
+  - Guides
+---
+
+A very cool thing about DDEV is that you don’t need to think about IP addresses or HTTPS or name resolution most of the time. Some of you will never need to. And you don't have to understand teh rest of this article to use DDEV.
+
+This article attempts to unwind what DDEV does for you, and what to do when things require a bit more effort.
+
+## Name resolution for browsers and URLs
+
+The first thing to understand is the structure of a URL used by your browser. A URL like `https://www.google.com/`/search?q=nothing` is structured with 3 main parts, the protocol (usually `https`), the hostname (like `www.google.com`), and the URI, (like `/search?q=nothing`). Every URL that your browser can visit has at least the protocol and the hostname. Sometimes the protocol is hidden implying the default `https`. The URI may be the default "front page".
+
+When your browser gets a URL, no matter how it gets it, and you want to display the page at that URL, the browser has to parse the URL, then turn the hostname into an IP address so it can find its way to it on the internet, then do the proper HTTP request to the internet site, including the URI. The key thing for us right now is that it has to look up that hostname.
```

Comment:
The sentence is lengthy. at first i thought make it two sentences but then i started to reducing it's content. a rough idea is the following: 

```suggestion
When your browser gets a URL, the browser has to parse the URL, turn the hostname into an IP address so it can find it on the internet, and  then do a proper HTTP request including the URI. The key take away for us here is that it has to look up that hostname.
```

Reply:
```suggestion
When your browser gets a URL, the browser has to parse the URL, turn the hostname into an IP address so it can find it on the internet, and then do a proper HTTP request including the URI. The key take-away for us is that it has to look up that hostname.
```